lib/features/composer/presentation/widgets/web/web_editor_widget.dart

```diff
@@ -101,6 +111,19 @@ class _WebEditorState extends State<WebEditorWidget> {
     if (_dropListener != null) {
       window.addEventListener("message", _dropListener!);
     }
+
+    _selectionChangeListener = (event) {
+      if (event is MessageEvent) {
+        final data = jsonDecode(event.data);
+
+        if (data['name'] == HtmlUtils.registerSelectionChangeListener.name) {
+          handleSelectionChange(data);
+        }
+      }
+    };
+    if (_selectionChangeListener != null) {
+      window.addEventListener("message", _selectionChangeListener!);
```

Comment:
We should listen for the `message` event in one place to avoid calling it multiple times. We could reuse `window.addEventListener("message", _dropListener!);`, but it should be modified appropriately for general use, as we might have multiple features that need to be handled in this function.